log/access.go

```diff
@@ -0,0 +1,78 @@
+package log
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"net"
+	"net/http"
+)
+
+type accessLogFormatter int
+
+type AccessEntry struct {
+	Request    *http.Request
+	Response   *http.Response
+	StatusCode int
+}
+
+func remoteAddr(r *http.Request) string {
+	// in case a proxy on the remote end sets the x-forwarded-for header,
+	// then we may get meaningless ip addresses here. Input needed. On,
+	// the other hand, Apache documentation,
+	// https://httpd.apache.org/docs/1.3/logs.html#common, says that it
+	// 'will be' the address of the proxy. If the proxy is the elb, or
+	// something on our end, then that's not very interesting. The go
+	// RemoteAddr field will be the proxy.
+
+	// can this contain multiple values? what's the meaning of it then?
+	// each proxy appends its own? no way to decide on the order.
+	ff := r.Header.Get("X-Forwarded-For")
+	if ff != "" {
+		return ff
+	}
+
+	return r.RemoteAddr
+}
+
+// strip port from addresses with hostname, ipv4 or ipv6
+func stripPort(address string) string {
+	h, _, err := net.SplitHostPort(address)
+	if err != nil {
+		h = ""
```

Comment:
Are you sure this is what you want to do? If address doesn't have a port it may still have a valid address...